sda/internal/database/db_functions.go

```diff
@@ -782,3 +782,73 @@ func (dbs *SDAdb) addKeyHash(keyHash, keyDescription string) error {
 
 	return nil
 }
+
+// ListDatasets lists all datasets as well as the status
+func (dbs *SDAdb) ListDatasets() ([]*DatasetInfo, error) {
+	dbs.checkAndReconnectIfNeeded()
+	db := dbs.DB
+
+	var datasets []*DatasetInfo
+	rows, err := db.Query("SELECT dataset_id,event,event_date FROM sda.dataset_event_log WHERE (dataset_id, event_date) IN (SELECT dataset_id,max(event_date) FROM sda.dataset_event_log GROUP BY dataset_id);")
```

Comment:
This might be more optimal, but maybe it ruins the readability:
```suggestion
	rows, err := db.Query("SELECT dataset_id, event, event_date FROM ( SELECT dataset_id, event, event_date, ROW_NUMBER() OVER (PARTITION BY dataset_id ORDER BY event_date DESC) AS rn FROM sda.dataset_event_log) AS LatestEvents WHERE rn = 1;")
```